vscode-smart-contracts/src/commands.ts

```diff
@@ -15,10 +19,155 @@ export async function version() {
     const version = await cargoConcordium.version();
     vscode.window.showInformationMessage(version);
   } catch (error) {
-    if (error instanceof ConfigError) {
+    if (error instanceof config.ConfigError) {
       vscode.window.showErrorMessage(error.message);
     } else {
-      throw error;
+      vscode.window.showErrorMessage("Unexpected error: " + error);
+    }
+  }
+}
+
+/**
+ * Run 'cargo-concordium build' using the directory of the currently focused editor.
+ * Printing the schema in base64 as part of stdout.
+ */
+export function build(editor: vscode.TextEditor) {
+  return buildWorker(editor, "stdout");
+}
+
+/**
+ * Run 'cargo-concordium build' using the directory of the currently focused editor.
+ * With no schema generation.
+ */
+export function buildSkipSchema(editor: vscode.TextEditor) {
+  return buildWorker(editor, "skip");
+}
+
+/**
+ * Run 'cargo-concordium build' using the directory of the currently focused editor.
+ * Embedding the schema into the resulting smart contract.
+ */
+export function buildEmbedSchema(editor: vscode.TextEditor) {
+  return buildWorker(editor, "embed");
+}
+
+/**
+ * Type representing the different settings for schema generation during cargo-concordium build.
+ */
+type SchemaSettings = "skip" | "embed" | "stdout";
+
+/**
+ * Internal worker for running 'cargo-concordium build' using the directory of the currently focused editor.
+ * Takes the schema setting as an argument.
+ */
+async function buildWorker(
+  editor: vscode.TextEditor,
+  schemaSettings: SchemaSettings
+) {
+  if (!(await haveWasmTargetInstalled())) {
+    const response = await vscode.window.showInformationMessage(
+      "The needed wasm32-unknown-unknown rust target seems to be missing. Should it be installed?",
+      "Install",
+      "Abort"
+    );
+    if (response === "Abort") {
+      vscode.window.showErrorMessage(
+        "Unable to build because of missing wasm32-unknown-unknown target."
+      );
+      return;
     }
+    await installWasmTarget();
   }
+  const cwd = path.dirname(editor.document.uri.fsPath);
+  const workspaceFolder = vscode.workspace.getWorkspaceFolder(
+    editor.document.uri
+  );
+  const schemaArgs =
+    schemaSettings === "skip"
+      ? []
+      : schemaSettings === "embed"
+      ? ["--schema-embed"]
+      : ["--schema-base64-out", "-"];
+  const additionalArgs = config.getAdditionalBuildArgs();
+  const args = schemaArgs.concat(additionalArgs);
```

Comment:
So whenever the command is executed, it will launch a VS Code task, which pops up a panel with the output.
Meaning, if they provide invalid arguments, they will see whatever error `cargo-concordium` outputs.